pkg/httpclient/url_resolver.go

```diff
@@ -0,0 +1,50 @@
+package httpclient
+
+import (
+	"fmt"
+	"github.com/mongodb-labs/pcgc/pkg/useful"
+	"net/url"
+	"path"
+)
+
+type baseURL struct {
+	base *url.URL
+}
+
+// URLResolver contract for resolving any paths against the given base URL
+type URLResolver interface {
+	Of(path string, v ...interface{}) string
+}
+
+// NewURLResolver builds a new API URL which can be used to build any path
+func NewURLResolver(base string) URLResolver {
+	return parseBaseURL(base)
+}
+
+// NewURLResolverWithPrefix builds a new API URL using a prefix for all other paths
+func NewURLResolverWithPrefix(base string, prefix string) URLResolver {
```

Comment:
Getting an error here but just wanted to point out that probably the word here is postfix and not prefix

Reply:
The term works in the context of any `path` that this object will later resolve will be prefixed with `prefix`